plugins/archlinux/archlinux.plugin.zsh

```diff
@@ -1,3 +1,130 @@
+#######################################
+#               Pacman                #
+#######################################
+
+# Pacman - https://wiki.archlinux.org/index.php/Pacman_Tips
+alias pacupg='sudo pacman -Syu'
+alias pacin='sudo pacman -S'
+alias pacins='sudo pacman -U'
+alias pacre='sudo pacman -R'
+alias pacrem='sudo pacman -Rns'
+alias pacrep='pacman -Si'
+alias pacreps='pacman -Ss'
+alias pacloc='pacman -Qi'
+alias paclocs='pacman -Qs'
+alias pacinsd='sudo pacman -S --asdeps'
+alias pacmir='sudo pacman -Syy'
+alias paclsorphans='sudo pacman -Qdt'
+alias pacrmorphans='sudo pacman -Rs $(pacman -Qtdq)'
+alias pacfileupg='sudo pacman -Fy'
+alias pacfiles='pacman -F'
+alias pacls='pacman -Ql'
+alias pacown='pacman -Qo'
+alias upgrade='sudo pacman -Syu'
+
+if (( $+commands[abs] && $+commands[aur] )); then
+  alias pacupd='sudo pacman -Sy && sudo abs && sudo aur'
+elif (( $+commands[abs] )); then
+  alias pacupd='sudo pacman -Sy && sudo abs'
+elif (( $+commands[aur] )); then
+  alias pacupd='sudo pacman -Sy && sudo aur'
+else
+  alias pacupd='sudo pacman -Sy'
+fi
+
+function paclist() {
+  # Based on https://bbs.archlinux.org/viewtopic.php?id=93683
+  pacman -Qqe | \
+    xargs -I '{}' \
+      expac "${bold_color}% 20n ${fg_no_bold[white]}%d${reset_color}" '{}'
```

Comment:
Is `expac` installed by default?

Reply:
No, it is in community repository. Although, I got it as a dependency of pacaur. Anyway, it is a very handy utility to query over pacman database.

Some moderator person on aforementioned thread replied that the script that I replaced did not make any sense, and you should be using `checkupdates` from `community/pacman-contrib` if you absolutely must. (before he proceeded to trash-talk oh-my-zsh, but that's a story for another time).

Honestly, this function is just-for-fun, and I don't see any real-world use cases, except for teaching newbies how to script Arch.

Reply:
+1